src/components/PasswordInput.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <label class="password-input">
+    Password
+    <div class="password-input__wrapper">
+      <input
+        class="password-input__input"
+        :type="type"
+        id="password"
+        name="password"
+        placeholder="Enter your password"
+      />
+      <button class="password-input__button" @click="showPassword">
+        <img
+          v-show="passwordHidden"
+          src="@/assets/icons/view.svg"
+          alt="show password"
+        />
+        <img
+          v-show="!passwordHidden"
+          src="@/assets/icons/view-off.svg"
+          alt="hide password"
+        />
+      </button>
+    </div>
+  </label>
+</template>
+
+<script>
+export default {
+  name: "PasswordInput",
+  components: {},
+  data() {
+    return {
+      type: "password",
+    };
+  },
+  computed: {
+    passwordHidden() {
+      return this.type === "password";
+    },
+  },
+  methods: {
+    showPassword() {
+      if (this.type === "password") {
+        this.type = "text";
+      } else {
+        this.type = "password";
+      }
+    },
+  },
```

Comment:
I guess that your code might be a bit shorter and cleaner if you change a bit your approach: I suggest you to keep `isPassowordHidden` in the data as a boolean value, and make `inputType` a computed value